tests/PHPStan/Parser/PhpDocParserTest.php

```diff
@@ -395,6 +397,89 @@ public function provideParamTagsData(): Iterator
 		];
 	}
 
+	public function provideTypelessParamTagsData(): Iterator
+	{
+		yield [
+			'OK',
+			'/** @param $foo description */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@param',
+					new TypelessParamTagValueNode(
+						false,
+						'$foo',
+						'description'
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK reference',
+			'/** @param &$foo description */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@param',
+					new TypelessParamTagValueNode(
+						false,
+						'$foo',
+						'description',
+						true
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK variadic',
+			'/** @param ...$foo description */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@param',
+					new TypelessParamTagValueNode(
+						true,
+						'$foo',
+						'description'
+					)
+				),
+			]),
+		];
+
+		yield [
+			'OK reference variadic',
+			'/** @param &...$foo description */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@param',
+					new TypelessParamTagValueNode(
+						true,
+						'$foo',
+						'description',
+						true
+					)
+				),
+			]),
+		];
+
+		yield [
+			'invalid without type and description',
```

Comment:
I think the conclusion of https://github.com/phpstan/phpdoc-parser/issues/77 is that `/** @param $foo */` shouldn't be a parse error either. The description can be an empty string, and if someone wants, they can have a rule to mark such tag as useless, but it would no longer be an objective bug to have such PHPDoc in the code...

So the grammar description + parser should be update to allow this tag.